### ISSUE TITLE
Unable to set headers in beforeRequest

### ISSUE THREAD

**Issue by mjnorman** [artilleryio/artillery#942]
According to closed issue #306 I should now be able to set headers properly in the beforeRequest method, specifically the Authorization header for oauth flows.  However, this does not appear to be the case.  Only the headers listed in the actual yml run file are added.  This was confirmed via wireshark.

Using version ---
```
Artillery: 1.6.1
Artillery Pro: not installed (https://artillery.io/pro)
Node.js: v11.6.0
OS: darwin/x64
```

PreRequest Script (redacted)
```js
var https = require('follow-redirects').https;
var fs = require('fs');
const { request } = require('http');

var qs = require('querystring');

module.exports = {
  beforeRequest: beforeRequest,
  printStatus: printStatus
};

function beforeRequest(requestParams, context, ee, next) {
  
  var options = {
    'method': 'POST',
    'hostname': '---',
    'path': '/as/token.oauth2',
    'headers': {
      'Content-Type': 'application/x-www-form-urlencoded'
    },
    'maxRedirects': 20
  };

  var req = https.request(options, function (res) {
    var chunks = [];

    res.on("data", function (chunk) {
      chunks.push(chunk);
    });

    res.on("end", function (chunk) {
      var body = Buffer.concat(chunks);
      var response = body.toString();
      var json = JSON.parse(response);
      var token = json.access_token;

      requestParams.headers.authorization = `Bearer ${token}`
      console.log(context)
      console.log(requestParams);
      console.log("==================================");
    });

    res.on("error", function (error) {
      console.error(error);
    });
  });

  var postData = qs.stringify({
    'client_id': '---',
    'client_secret': '---',
    'grant_type': 'client_credentials',
	'scope': '---'
  });

  req.write(postData);

  req.end();
```

run file ---
```js
config:
  target: "https://--"
  phases:
    - duration: 300
      arrivalRate: 2 
  processor: "./oauth.js"

scenarios:
  - flow:
      - get:
          url: "--"
          qs:
            parameters: "parameters"
          headers:
            Content-Type: "application/json"
            Accept: application/json
          json: {}
          beforeRequest: "beforeRequest"
```

The workaround of setting the token externally to an environment variable, and adding the header such as below works, but is inconvenient for short lived tokens.

```json
          headers:
            Authorization: "Bearer {{ $processEnvironment.SERVICE_API_KEY }}"
```

**Comment by hassy** [comment]
Yep, definitely inconvenient! To make your function work you need to return control to Artillery by calling `next()` from your function, in the `"end"` event handler after setting the new header.

**Comment by mjnorman** [comment]
The function flow works fine it just doesn't set the header before proceeding.  The scenario itself runs correctly.  I put a next() in there anyway, did not make a difference.

**Comment by hassy** [comment]
I put together a working example of using `beforeRequest` to set a custom header here:

https://github.com/artilleryio/artillery-examples/tree/main/http-set-custom-header

Give it a go and lmk how you get on.

**Comment by mjnorman** [comment]
Thank you for the example.  I think the only thing I had wrong was the `return next()`; in the wrong place.  What I didn't show in the example above was that I had 
```js
  req.write(postData);
  req.end();
  return next();
```

and instead I moved it to `res.on('data')` function.  Earlier I had just added it into `res.on('data')` and did not remove the extraneous one at the end of the header function.  

Thanks for your quick feedback!
